python/ray/rllib/agents/agent.py

```diff
@@ -227,7 +227,7 @@ def compute_action(self, observation, state=None, policy_id="default"):
             observation, update=False)
         return self.local_evaluator.for_policy(
             lambda p: p.compute_single_action(
-                filtered_obs, state, is_training=False)[0],
+                filtered_obs, state, is_training=False),
```

Comment:
I think this will break some unit tests (and also is a large behavior change).

How about this: we return [0] if state=None, otherwise we return all the values? That way, you get the rnn state if you need it.

Reply:
Can we also add this to the method doc?

Reply:
Sounds good, will add this now.